site/docs/integrations/astro.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Astro
+parent: integrations
+---
+
+# Astro
+
+A plugin for integrating vanilla-extract with [Astro](https://astro.build).
+
+## Installation
+
+```bash
+npm install @vanilla-extract/css
+npm install @vanilla-extract/vite-plugin --save-dev
+```
+
+## Setup
+
+Add Vanilla Extract Vite plugin to the Astro configuration:
+
+```js
+import { defineConfig } from 'astro/config';
+import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
+
+// https://astro.build/config
+export default defineConfig({
+  vite: {
+    plugins: [vanillaExtractPlugin()]
+  }
+});
+```
+
+You'll then be able to use `style` and other APIs in `.css.ts` files.
+
+```ts
+// button.css.ts
+
+import { style } from '@vanilla-extract/css';
+
+export const button = style({
+  padding: '0.5rem 1rem',
+  border: 'none',
+  borderRadius: '0.25rem',
+  color: 'white',
+  background: '#333'
+});
+```
+
+```astro
+---
+// Button.astro
+import { button } from './button.css'
+---
+
+<button class={button}>Click Me!</button>
+```
```

Comment:
This shows up as unstyled for me. 😬 
![image](https://github.com/vanilla-extract-css/vanilla-extract/assets/3335181/cf3a0236-fc10-47bf-be95-33a3be4141c2)

I took a peek at the `CompiledCode` and `SyntaxHighlighter` system but didn't want to make too many changes. I was scared. What would you all advise be done here?

FWIW the progress I made was trying to import and register the `markup` syntax highlighting language for Astro, but that didn't get all the way:

```ts
// @ts-ignore
import markup from 'react-syntax-highlighter/dist/esm/languages/prism/markup';
SyntaxHighlighter.registerLanguage('astro', markup);
```

Reply:
After a bit of digging, it looks like it will take a lot of work to have syntax highlighting for Astro files. Given it's only used on this page, I think we're ok with not having it.